### PR TITLE
Updates misleading example on ONVK additional network config

### PR DESCRIPTION
**Pull request by stevsmit**
Version(s):
4.13

Issue:
https://issues.redhat.com/browse/OSDOCS-5818

Link to docs preview:
https://59219--docspreview.netlify.app/openshift-enterprise/latest/networking/multiple_networks/configuring-additional-network.html#configuration-ovnk-network-plugin-json-object_configuring-additional-network

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by maiqueb — APPROVED** [review]
Perfect ! 

Thank you.

**Comment by maiqueb** [comment]
@jcaamano do you think this is clear ?

File: modules/configuration-ovnk-network-plugin-json-object.adoc [hunk 1]
```diff
@@ -37,7 +37,10 @@ For `"topology":"layer2"` deployments, IPv6 (`2001:DBB::/64`) and dual-stack (`1
 
 |`netAttachDefName`
 |`string`
-| The namespace and the network attachment name. The value must match the values specified for `namespace` and `name` when defining the network attachment definition object. For example, `ns1/l2-network` refers to the `ns1` namespace and the `l2-network` network attachment definition name.
+| The network attachment name. These networks are not namespaced. For example, you can have one `netAttachDefName` named `l2-network` on one `namespace` object, and a second `netAttachDefName` named `l2-network` on a second `namespace` object. This ensures that pods on different namespaces can communicate over a secondary overlay network.
+
```
Comment (jcaamano) [inline modules/configuration-ovnk-network-plugin-json-object.adoc:41]: I find that the previous doc was fine as Network attachment definitions are namespaced.

I think this should go in the `name` row, not the `netAttachDefName` row.

**Review by jcaamano — CHANGES_REQUESTED** [review]

File: modules/configuration-ovnk-network-plugin-json-object.adoc [hunk 2]
```diff
@@ -37,7 +37,10 @@ For `"topology":"layer2"` deployments, IPv6 (`2001:DBB::/64`) and dual-stack (`1
...
+
+The value must match the values specified for the `name` when defining the network attachment definition object.
```
Comment (jcaamano) [inline modules/configuration-ovnk-network-plugin-json-object.adoc:43]:
```
The value must match the namespace and name of the network attachment definition where this configuration is included.
```

File: modules/configuration-ovnk-network-plugin-json-object.adoc (hunk 1 above)
Comment (maiqueb) [reply modules/configuration-ovnk-network-plugin-json-object.adoc:41]: I agree w/ @jcaamano , good catch. We were making the docs worse. Good catch. I apologize for lgtm'ing this.

I think the clarification should indeed happen on the `name` attribute - this is the attribute that should **not** have anything to do with namespaces.

**Review by maiqueb — CHANGES_REQUESTED** [review]
Sorry for having previously approved.

File: modules/configuration-ovnk-network-plugin-json-object.adoc [hunk 3]
```diff
@@ -15,7 +15,7 @@ The following table describes the configuration parameters for the OVN-Kubernete
 
 |`name`
 |`string`
-|The name of the network. This value must be unique across all `NetworkAttachmentDefinitions`. Failure to make this attribute unique might result in the merging of two networks with the same name. Merging two networks by sharing similar names is unsupported.
+|The network attachment name. These networks are not namespaced. For example, you can have one `netAttachDefName` named `l2-network` on one `namespace` object, and a second `netAttachDefName` named `l2-network` on a second `namespace` object. This ensures that pods on different namespaces can communicate over a secondary overlay network.
```
Comment (maiqueb) [inline modules/configuration-ovnk-network-plugin-json-object.adoc:18]:
```suggestion
|The name of the network. These networks are not namespaced. For example, you can have one `netAttachDefName` named `l2-network` on one `namespace` object, and a second `netAttachDefName` named `l2-network` on a second `namespace` object. This ensures that pods on different namespaces can communicate over a secondary overlay network.
```

**Review by maiqueb — CHANGES_REQUESTED** [review]
I think this read better before.

**Review by maiqueb — APPROVED** [review]
Thank you for your patience and for the fix.

File: modules/configuration-ovnk-network-plugin-json-object.adoc [hunk 4]
```diff
@@ -15,7 +15,7 @@ The following table describes the configuration parameters for the OVN-Kubernete
 
 |`name`
 |`string`
-|The name of the network. This value must be unique across all `NetworkAttachmentDefinitions`. Failure to make this attribute unique might result in the merging of two networks with the same name. Merging two networks by sharing similar names is unsupported.
+|The name of the network. These networks are not namespaced. For example, you can have one `netAttachDefName` named `l2-network` on one `namespace` object, and a second `netAttachDefName` named `l2-network` on a second `namespace` object. This ensures that pods on different namespaces can communicate over a secondary overlay network.
```
Comment (jcaamano) [inline modules/configuration-ovnk-network-plugin-json-object.adoc:18]: This is a bit more nuanced so I am suggesting the alternative:
```
The name of the network. These networks are not namespaced. For example, you can have a network named 
`l2-network` referenced from two different NetworkAttachmentDefinition that exist on two different
namespaces. This ensures that pods making use of NetworkAttachmentDefinition on their own different
namespaces can communicate over the same secondary network. However, those two different
NetworkAttachmentDefinition should also share the same network specific parameters like `topology`,
`subnets`, `mtu`, `excludeSubnets`, etc.
```

File: modules/configuration-ovnk-network-plugin-json-object.adoc [hunk 5]
```diff
@@ -38,6 +38,8 @@ For `"topology":"layer2"` deployments, IPv6 (`2001:DBB::/64`) and dual-stack (`1
 |`netAttachDefName`
 |`string`
 | The namespace and the network attachment name. The value must match the values specified for `namespace` and `name` when defining the network attachment definition object. For example, `ns1/l2-network` refers to the `ns1` namespace and the `l2-network` network attachment definition name.
++
+The value must match the `namespace` and metadata name of the network attachment definition object where this configuration is included.
```
Comment (jcaamano) [inline modules/configuration-ovnk-network-plugin-json-object.adoc:42]: I just notice that are just saying the same thing in line 40 above. I prefer this version but embed in the paragraph. 

So I would rephrase the whole thing to: 
```
The metadata `namespace` and `name` of the network attachment definition object where this
configuration is included. For example, if this configuration is defined in a network attachment
definition in namespace `ns1` named `l2-network`, this should be set to `ns1/l2-network`.
```

File: modules/configuration-ovnk-network-plugin-json-object.adoc [hunk 6]
```diff
@@ -15,7 +15,11 @@ The following table describes the configuration parameters for the OVN-Kubernete
 
 |`name`
 |`string`
-|The name of the network. This value must be unique across all `NetworkAttachmentDefinitions`. Failure to make this attribute unique might result in the merging of two networks with the same name. Merging two networks by sharing similar names is unsupported.
+|The name of the network. These networks are not namespaced. For example, you can have a network named
+`l2-network` referenced from two different `NetworkAttachmentDefinitions` that exist on two different
+namespaces. This ensures that pods making use of the `NetworkAttachmentDefinition` on their own different
+namespaces can communicate over the same secondary network. However, those two different `NetworkAttachmentDefinitions` should also share the same network specific parameters like `topology`,
```
Comment (jboxman-rh) [inline modules/configuration-ovnk-network-plugin-json-object.adoc:21]: However, those two different `NetworkAttachmentDefinitions` must also share the same network specific parameters such as `topology`, `subnets`, `mtu`, and `excludeSubnets`.

**Review by jboxman-rh — COMMENTED** [review]
@stevsmit, left a comment, thanks!

File: modules/configuration-ovnk-network-plugin-json-object.adoc [hunk 7]
```diff
@@ -37,7 +41,9 @@ For `"topology":"layer2"` deployments, IPv6 (`2001:DBB::/64`) and dual-stack (`1
 
 |`netAttachDefName`
 |`string`
-| The namespace and the network attachment name. The value must match the values specified for `namespace` and `name` when defining the network attachment definition object. For example, `ns1/l2-network` refers to the `ns1` namespace and the `l2-network` network attachment definition name.
+|The metadata `namespace` and `name` of the network attachment definition object where this
+configuration is included. For example, if this configuration is defined in a network attachment
+definition in namespace `ns1` named `l2-network`, this should be set to `ns1/l2-network`.
```
Comment (jcaamano) [inline modules/configuration-ovnk-network-plugin-json-object.adoc:46]: nit: should we use `'NetworkAttachmentDefinition'` instead of `network attachment
definition` as elsewhere?

File: modules/configuration-ovnk-network-plugin-json-object.adoc (hunk 6 above)
Comment (stevsmit) [reply modules/configuration-ovnk-network-plugin-json-object.adoc:21]: Fixed. Thanks @jboxman-rh

Comment (jcaamano) [reply modules/configuration-ovnk-network-plugin-json-object.adoc]: I am not reading it as @jboxman-rh phrased it. I assumed using **MUST** was an important part of that re-phrase.

Comment (jcaamano) [reply modules/configuration-ovnk-network-plugin-json-object.adoc]: What I mean here is where it says:
```
However, those two different `NetworkAttachmentDefinitions` should also share the same network specific parameters like `topology`, `subnets`, `mtu`, and `excludeSubnets`.
```
should say
```
However, those two different NetworkAttachmentDefinitions must also share the same network specific parameters such as topology, subnets, mtu, and excludeSubnets.
```
as per @jboxman-rh suggestion above

Comment (stevsmit) [reply modules/configuration-ovnk-network-plugin-json-object.adoc]: Got it @jboxman-rh @jcaamano

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM
